components/raftstore-v2/src/operation/command/write/simple_write.rs

```diff
@@ -53,19 +66,23 @@ impl SimpleWriteReqEncoder {
             buf,
             channels: vec![],
             size_limit,
+            write_type: bin.write_type,
             notify_proposed,
         }
     }
 
-    /// Encode the simple write into the buffer dispite header check.
+    /// Encode the simple write into the buffer.
     ///
     /// Return false if the buffer limit is reached or the write can be amended.
     #[inline]
     pub fn amend(&mut self, header: &RaftRequestHeader, bin: &SimpleWriteBinary) -> bool {
         if *self.header != *header {
             return false;
         }
-        if self.buf.len() + bin.buf.len() < self.size_limit {
+        if self.write_type == bin.write_type
+            && bin.write_type != WriteType::Unspecified
```

Comment:
Seems redundant.

Reply:
It can be `Unspecified` when the binary is frozen.

Reply:
If self.type is unspecified, doesn't that means this batch is empty and can be amended?

Reply:
Check https://github.com/tikv/tikv/pull/14294/files#diff-271e125f518c42e15916957309f9cbb74b1f248836a1bb73370cfca514d45125R28-R32

Reply:
I don't think you are paying attention. I'm saying, when self.write_type == bin.write_type == Unspecified, this branch should be okay to execute, because `SimpleWriteEncoder` is empty (created by `with_capacity`). But then I looked closer, apparently `SimpleWriteEncoder::new` will introduce the case where self.type is unspecified but non-empty at the same time. So my question answered.